velox/common/base/SuccinctPrintUtil.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iomanip>
+#include <sstream>
+
+#include "velox/common/base/SuccinctPrintUtil.h"
+
+namespace facebook::velox::printer {
+static constexpr std::string_view timeUnits[4] = {"ns", "us", "ms", "s"};
+static constexpr std::string_view byteUnits[4] = {"B", "KB", "MB", "GB"};
+static int kTimeScale = 1000;
+static int kByteScale = 1024;
+
+namespace {
+std::string succinctPrint(
+    uint64_t value,
+    const std::string_view units[],
```

Comment:
const &

Reply:
This syntax is essentially passing by a pointer. I will use a pointer to avoid confusion.